temporal-sdk/src/main/java/io/temporal/client/ListScheduleListDescriptionIterator.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (C) 2022 Temporal Technologies, Inc. All Rights Reserved.
+ *
+ * Copyright (C) 2012-2016 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Modifications copyright (C) 2017 Uber Technologies, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this material except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.temporal.client;
+
+import com.google.protobuf.ByteString;
+import io.temporal.api.schedule.v1.ScheduleListEntry;
+import io.temporal.api.workflowservice.v1.ListSchedulesRequest;
+import io.temporal.api.workflowservice.v1.ListSchedulesResponse;
+import io.temporal.internal.client.external.GenericWorkflowClient;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+public final class ListScheduleListDescriptionIterator
```

Comment:
I'd agree, but this really should not be public , but it is hard to avoid. The TLDR version is either `ListScheduleListDescriptionIterator` needs to be public of we make `EagerPaginator` public or we refactor a bunch of how the Java SDK is layed out.